zebra/zapi_msg.c

```diff
@@ -2962,6 +2962,41 @@ static inline void zread_iptable(ZAPI_HANDLER_ARGS)
 	return;
 }
 
+static inline void zread_neigh_discover(ZAPI_HANDLER_ARGS)
+{
+	struct stream *s;
+	ifindex_t ifindex;
+	struct interface *ifp;
+	struct prefix p;
+	struct ipaddr ip;
+
+	s = msg;
+
+	STREAM_GETL(s, ifindex);
+
+	ifp = if_lookup_by_index_per_ns(zvrf->zns, ifindex);
+	if (!ifp) {
+		zlog_debug("Failed to lookup ifindex: %u", ifindex);
+		return;
+	}
+
+	STREAM_GETC(s, p.family);
+	STREAM_GETC(s, p.prefixlen);
```

Comment:
nit: the prefix length isn't being used for anything.

Since we only care about the IP address, I think it would be better from an API perspective to change `zclient_send_neigh_discovery_req()` to accept an `struct ipaddr *` parameter and use `stream_put_ipaddr()`/`stream_get_ipaddr()` to encode/decode it respectively.

Reply:
I specifically asked for switching over to `struct prefix` for both ipv6 support in the kernel and I wanted a unified way to pass data down that was uniform across FRR.  I do not agree with passing the `struct ipaddr `

Reply:
ipv6 support in the future

Reply:
Fair enough. My approval still stands :)